packages/core-mobile/app/services/walletconnectv2/WalletConnectService.ts

```diff
@@ -196,45 +200,48 @@ class WalletConnectService implements WalletConnectServiceInterface {
     chainId: number
     account: CorePrimaryAccount
   }): Promise<void> => {
-    if (isBitcoinChainId(chainId)) {
-      Logger.info('skip updating WC session for bitcoin network')
-      return
+    const topic = session.topic
```

Comment:
had to refactor a little bit to make wallet connect work smoothly with all the non-evm namespaces. there should be no error updating sessions now and whenever you switch to another network/account, sessions should be updated correctly.